### ISSUE TITLE
Style/SafeNavigation false positives when second AND operand is negated

### ISSUE THREAD

**Issue by bbascarevic-tti** [bbatsov/rubocop#4835]
The [Style/SafeNavigation] cop incorrectly flags this code as optimizable using safe navigation operator:

```
if foo && !foo.empty?
```

Auto-correct changes it to this:
```
if !foo&.empty?
```

and ultimately this:
```
unless foo&.empty?
```

...which is an altogether different behavior, since it will evaluate to true instead of false if `foo` is nil.

--------

## Expected behavior

Rubocop should not flag this as a valid safe navigation operator candidate, since `&&` is used as a normal short-circuit `AND`, not as a nil guard (evident by the negated part after `&&`).

## Actual behavior

Rubocop flags this as a valid safe navigation operator candidate, and even auto-corrects it wrong.

## Steps to reproduce the problem

```
s = nil
puts 'Should not see me #1' unless s&.empty?
puts 'Should not see me #2' if s && !s.empty?
```

## RuboCop version
```
$ rubocop -V
0.50.0 (using Parser 2.4.0.0, running on ruby 2.3.3 x86_64-linux)
```

**Comment by Drenmi** [comment]
Duplicate of #4745. Fixed in #4752.
